ape_solidity/compiler.py

```diff
@@ -104,25 +136,52 @@ def available_versions(self) -> List[Version]:
     def installed_versions(self) -> List[Version]:
         return solcx.get_installed_solc_versions()
 
-    def get_import_remapping(self, base_path: Optional[Path] = None) -> Dict[str, str]:
+    def get_import_remapping(
+        self, source_paths: Optional[Set[Path]] = None, base_path: Optional[Path] = None
+    ) -> Dict[str, str]:
         """
         Specify the remapping using a ``=`` separated str
         e.g. ``'@import_name=path/to/dependency'``.
         """
+        base_path = base_path or self.project_manager.contracts_folder
         import_map: Dict[str, str] = {}
         items = self.config.import_remapping
 
         if not items:
             return import_map
 
-        if not isinstance(items, (list, tuple)) or not isinstance(items[0], str):
+        elif source_paths:
+            # Filter out unused import remappings
+            config_mapping = []
+            import_lines = _get_import_lines([p for p in source_paths])
```

Comment:
Thoughts about passing the set instead of converting to a list?

Reply:
Done!